bitnami/redis/templates/configmap-scripts.yaml

```diff
@@ -294,28 +294,79 @@ data:
+    REDIS_SERVICE="{{ include "redis.fullname" . }}.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}"
+    
+    if [[ -n "$REDIS_PASSWORD" ]]; then
+      export REDISCLI_AUTH="$REDIS_PASSWORD"
+    fi
     if [[ -n "$REDIS_PASSWORD_FILE" ]]; then
       password_aux=$(cat "$REDIS_PASSWORD_FILE")
-      export REDIS_PASSWORD="$password_aux"
+      export REDISCLI_AUTH="$password_aux"
     fi
 
-    if is_boolean_yes "$REDIS_SENTINEL_TLS_ENABLED"; then
-      sentinel_info_command="redis-cli {{- if .Values.usePassword }} -a ${REDIS_PASSWORD} {{- end }} -h ${REDIS_SERVICE} -p {{ .Values.sentinel.port }} --tls --cert ${REDIS_SENTINEL_TLS_CERT_FILE} --key ${REDIS_SENTINEL_TLS_KEY_FILE} --cacert ${REDIS_SENTINEL_TLS_CA_FILE} sentinel get-master-addr-by-name {{ .Values.sentinel.masterSet }}"
+    if am_i_the_master; then
+      run_sentinel_command failover "{{ .Values.sentinel.masterSet }}"
+      for i in $(seq 10); do
+        if ! am_i_the_master; then
+          exit 0
+        fi
+        echo "Waiting for failover to finish"
+        sleep 1
+      done
     else
-      sentinel_info_command="redis-cli {{- if .Values.usePassword }} -a ${REDIS_PASSWORD} {{- end }} -h ${REDIS_SERVICE} -p {{ .Values.sentinel.port }} sentinel get-master-addr-by-name {{ .Values.sentinel.masterSet }}"
+      exit 0
     fi
-    REDIS_SENTINEL_INFO=($($sentinel_info_command))
-    REDIS_MASTER_HOST="${REDIS_SENTINEL_INFO[0]}"
+  prestop-redis.sh: |
+    #!/bin/bash
 
-    if [[ "$REDIS_MASTER_HOST" == "$(hostname -i)" ]]; then
-      if is_boolean_yes "$REDIS_SENTINEL_TLS_ENABLED"; then
-        redis-cli {{- if .Values.usePassword }} -a "$REDIS_PASSWORD" {{- end }} -h "$REDIS_SERVICE" -p {{ .Values.sentinel.port }} --tls --cert "$REDIS_SENTINEL_TLS_CERT_FILE" --key "$REDIS_SENTINEL_TLS_KEY_FILE" --cacert "$REDIS_SENTINEL_TLS_CA_FILE" sentinel failover mymaster
+    . /opt/bitnami/scripts/libvalidations.sh
+    
+    run_redis_command() {
+      if is_boolean_yes "$REDIS_TLS_ENABLED"; then
+        redis-cli -h 127.0.0.1 -p ${REDIS_TLS_PORT} --tls --cert ${REDIS_TLS_CERT_FILE} --key ${REDIS_TLS_KEY_FILE} --cacert ${REDIS_TLS_CA_FILE} "$@"
       else
-        redis-cli {{- if .Values.usePassword }} -a "$REDIS_PASSWORD" {{- end }} -h "$REDIS_SERVICE" -p {{ .Values.sentinel.port }} sentinel failover mymaster
+        redis-cli -h 127.0.0.1 -p ${REDIS_PORT} "$@"
       fi
+    }
+    am_i_the_master() {
+      REDIS_ROLE=$(run_redis_command role | head -1)
+
+      [[ "$REDIS_ROLE" == "master" ]]
+    }
+
+    if [[ -n "$REDIS_PASSWORD" ]]; then
+      export REDISCLI_AUTH="$REDIS_PASSWORD"
+    fi
+    if [[ -n "$REDIS_PASSWORD_FILE" ]]; then
+      password_aux=$(cat "$REDIS_PASSWORD_FILE")
+      export REDISCLI_AUTH="$password_aux"
+    fi
+
+    if am_i_the_master; then
+      echo "Waiting for sentinel to run failover for up to 10s"
+      for i in $(seq 10); do
+        if ! am_i_the_master; then
+          exit 0
+        fi
+        sleep 1
+      done
```

Comment:
Try to use here a similar approach to the above one using `retry_while`

Reply:
Is there a list of these prepared functions with descriptions somewhere? I can definitely clean up the loop with it but I am not sure where to find the function definitions/docs.

Reply:
You can check them here: https://github.com/bitnami/bitnami-docker-redis/tree/master/6.2/debian-10/prebuildfs/opt/bitnami/scripts